source/distributed/tria.cc

```diff
@@ -3267,64 +3265,59 @@ namespace parallel
 
-      // note that now we have some content in the p4est objects and call the
-      // functions that do the actual work (which are dimension dependent, so
-      // separate)
-      triangulation_has_content = true;
+        try
+          {
+            create_triangulation(std::get<0>(coarse_mesh_description),
+                                 std::get<1>(coarse_mesh_description),
+                                 std::get<2>(coarse_mesh_description));
+          }
+        catch (const typename Triangulation<dim>::DistortedCellList &)
+          {
+            // the underlying triangulation should not be checking for distorted
+            // cells
+            Assert(false, ExcInternalError());
+          }
 
-      if (const dealii::parallel::distributed::Triangulation<dim, spacedim>
-            *other_tria_x =
-              dynamic_cast<const dealii::parallel::distributed::
-                             Triangulation<dim, spacedim> *>(&other_tria))
-        {
-          coarse_cell_to_p4est_tree_permutation =
-            other_tria_x->coarse_cell_to_p4est_tree_permutation;
-          p4est_tree_to_coarse_cell_permutation =
-            other_tria_x->p4est_tree_to_coarse_cell_permutation;
-          this->cell_attached_data = other_tria_x->cell_attached_data;
-          this->data_transfer      = other_tria_x->data_transfer;
-
-          settings = other_tria_x->settings;
-        }
-      else
+        for (const auto i : other_tria.get_manifold_ids())
+          if (i != numbers::flat_manifold_id)
+            this->set_manifold(i, other_tria.get_manifold(i));
+      }
+
+      // deal with parallel distributed specifics
+      if (const auto *other_distributed =
+            dynamic_cast<const dealii::parallel::distributed::
+                           Triangulation<dim, spacedim> *>(&other_tria))
         {
-          setup_coarse_cell_to_p4est_tree_permutation();
-        }
+          // copy attributes
+          this->cell_attached_data = other_distributed->cell_attached_data;
+          this->data_transfer      = other_distributed->data_transfer;
 
-      copy_new_triangulation_to_p4est(std::integral_constant<int, dim>());
+          settings = other_distributed->settings;
 
-      try
-        {
-          copy_local_forest_to_triangulation();
+          // recover refinement tree
+          load(other_distributed->get_p4est());
         }
-      catch (const typename Triangulation<dim>::DistortedCellList &)
+
+      // release unused vector memory because we will have very different
+      // vectors now
+      if (dynamic_cast<const dealii::parallel::TriangulationBase<dim, spacedim>
+                         *>(&other_tria))
         {
-          // the underlying triangulation should not be checking for distorted
-          // cells
-          Assert(false, ExcInternalError());
+          ::dealii::internal::GrowingVectorMemoryImplementation::
+            release_all_unused_memory();
```

Comment:
This call was present in `parallel::TriangulationBase::copy_triangulation`. I don't know if it's necessary in this context, but it probably won't hurt anyways.